src/fidesctl/core/config/credentials_settings.py

```diff
@@ -0,0 +1,111 @@
+"""This module handles finding and parsing fides configuration files."""
+
+# pylint: disable=C0115,C0116, E0213
+import os
+from typing import Dict, Optional
+
+from pydantic import BaseModel
+
+
+# TODO: These credentials objects could be shared with api code being written in fides#642
+class DatabaseCredentials(BaseModel):
+    """Class used to validate database credentials fields when retrieved from config"""
+
+    connection_string: str
+
+
+class OktaCredentials(BaseModel):
+    """Class used to validate okta credentials fields when retrieved from config"""
+
+    # camel case matches okta client config model
+    orgUrl: str
+    token: str
+
+
+class AwsCredentials(BaseModel):
+    """Class used to validate aws credentials fields when retrieved from config"""
+
+    aws_access_key_id: str
+    aws_secret_access_key: str
+    region_name: str
+
+
+CREDENTIALS_ENV_PREFIX = "FIDESCTL__CREDENTIALS__"
+NESTED_KEY_DELIMITER = "__"
+
+
+def merge_credentials_environment(credentials_dict: Dict) -> Dict:
+    """
+    Given a dict of config settings, merges configs which are found
+    in environment variables. Environment variable configs are treated
+    as overrides. Environment variables muse use the prefix
+    FIDESCTL__CREDENTIALS__ and use the delimiter __ to override nested keys.
+    Example:
+    FIDESCTL__CREDENTIALS__POSTGRES_1_CONNECTION_STRING="MY_CONN"
+
+    Merges:
+    {"postgres_1" : { "connection_string": "MY_CONN"}}
+    """
+    for key, value in os.environ.items():
+        if key.startswith(CREDENTIALS_ENV_PREFIX):
+            # TODO Replace with removeprefix when 3.9 is required - https://peps.python.org/pep-0616
+            # keys are not case sensitive and will be matched with lower case equivalent in config
+            stripped_key = key[len(CREDENTIALS_ENV_PREFIX) :].lower()
+            insert_environment(
+                current_dict=credentials_dict, env_key=stripped_key, env_val=value
+            )
+    return credentials_dict
```

Comment:
Just wanted to make sure I understand this, but this covers the use case of using credential IDs without having to store secrets in the `.toml` itself, correct? As in, we can define most of a configuration block that is not secret and override the rest with environment variables?

I still need to look through the docs changes more closely as well, but this could almost be worth a user guide of sorts. Can imagine this being a common use case for anyone using credential IDs

Reply:
Yeah that's right. You can just override anything as needed or provide the full config through environment variables.

I think this test covers the use case you're wondering about:
```
@patch.dict(
    os.environ,
    {
        "FIDESCTL__CREDENTIALS__POSTGRES_1__CONNECTION_STRING": "postgresql+psycopg2://fidesctl:env_variable.com:5439/fidesctl_test",
        "FIDESCTL__CREDENTIALS__AWS_ACCOUNT_1__ACCESS_KEY_ID": "ACCESS_KEY_ID_1",
        "FIDESCTL__CREDENTIALS__AWS_ACCOUNT_1__ACCESS_KEY": "ACCESS_KEY_1",
    },
    clear=True,
)
@pytest.mark.unit
def test_merge_credentials_environment_mixed_configs() -> None:
    "Test that merging environment variables works with mixed configs"
    credentials_dict = {"aws_account_1": {"region": "us-east-1"}}
    merge_credentials_environment(credentials_dict)

    assert credentials_dict == {
        "postgres_1": {
            "connection_string": "postgresql+psycopg2://fidesctl:env_variable.com:5439/fidesctl_test"
        },
        "aws_account_1": {
            "region": "us-east-1",
            "access_key_id": "ACCESS_KEY_ID_1",
            "access_key": "ACCESS_KEY_1",
        },
    }
```

A guide would be nice. Open to creating a new issue for that since I don't think it's a priority right now.

Reply:
Sounds good! Ya I think it would help, referenced probably a follow on issue in the approval comments too 👍🏽